lib_eio_linux/eio_linux.ml

```diff
@@ -1065,6 +1068,29 @@ class dir fd = object
   method mkdir ~perm path =
     Low_level.mkdir_beneath ~perm ?dir:fd path
 
+  method read_dir path =
+    let rec read_all acc fd =
+      try
+        read_all (Unix.readdir fd :: acc) fd
+      with End_of_file -> List.rev acc |> List.filter (function ".." | "." -> false | _ -> true)
+    in
+    Switch.run (fun sw ->
+      (* Using openat2 to preserve the permission checking, but not using the returned
+         file descriptor. Uring doesn't support opendir. *)
+      let dir =
+        wrap_errors path @@ fun () ->
+        Low_level.openat2 ~sw ~seekable:false ?dir:fd path
+          ~access:`R
+          ~flags:Uring.Open_flags.(cloexec + path + directory)
```

Comment:
```suggestion
          ~flags:Uring.Open_flags.(cloexec + directory)
```
Using `path` is OK for the code as it is, but to avoid opening it twice we'll need to remove this.